[Bug]: litellm package missing from default group in pyproject.toml

### Contact Details

_No response_

### What happened?

When installing the latest and greatest version of orchestrator-core, it fails to run our orchestrator since this import chain fails:

```python
gso/workflows/prefix_list/validate_prefix_list.py:8: in <module>
    from orchestrator.workflows.utils import validate_workflow
.venv/lib/python3.12/site-packages/orchestrator/workflows/utils.py:33: in <module>
    from orchestrator.workflows.steps import (
.venv/lib/python3.12/site-packages/orchestrator/workflows/steps.py:23: in <module>
    from orchestrator.search.indexing import run_indexing_for_entity
.venv/lib/python3.12/site-packages/orchestrator/search/indexing/__init__.py:14: in <module>
    from .tasks import run_indexing_for_entity
.venv/lib/python3.12/site-packages/orchestrator/search/indexing/tasks.py:20: in <module>
    from orchestrator.search.indexing.indexer import Indexer
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
(...)
>   from litellm.utils import encode, get_max_tokens
E   ModuleNotFoundError: No module named 'litellm'
```

As the `litellm` is not included in the pyproject.toml file as a default dependency, it does not get installed when running `uv sync`. Although we have the new search and agent disabled, so I wouldn't expect to need to install this dependency.

### Version

4.5.0

### What python version are you seeing the problem on?

_No response_

### Relevant log output

```shell

```

--- a/orchestrator/workflows/steps.py
+++ b/orchestrator/workflows/steps.py
@@ -19,8 +19,6 @@ from orchestrator import llm_settings
 from orchestrator.db import db
 from orchestrator.db.models import ProcessSubscriptionTable
 from orchestrator.domain.base import SubscriptionModel
-from orchestrator.search.core.types import EntityType
-from orchestrator.search.indexing import run_indexing_for_entity
 from orchestrator.services.settings import reset_search_index
 from orchestrator.services.subscriptions import get_subscription
 from orchestrator.targets import Target
@@ -157,6 +155,9 @@ def refresh_subscription_search_index(subscription: SubscriptionModel | None) ->
     try:
         reset_search_index()
         if llm_settings.SEARCH_ENABLED and subscription:
+            from orchestrator.search.core.types import EntityType
+            from orchestrator.search.indexing import run_indexing_for_entity
+
             run_indexing_for_entity(EntityType.SUBSCRIPTION, str(subscription.subscription_id))
     except Exception:
         # Don't fail workflow in case of unexpected error

